src/lib.rs

```diff
@@ -6,31 +6,57 @@
 //! Implementations are available that accept a fixed length input
 //! while returning a variable length output, and vice versa.
 //!
-//! ## Asynchronous resampling
+//! ### Input and output data format
+//!
+//! Input and output data is stored non-interleaved.
+//!
+//! The output data is stored in a vector or vectors, `Vec<Vec<f32>>` or `Vec<Vec<f64>>`.
+//! The inner vectors (`Vec<f32>` or `Vec<f64>`) hold the sample values for one channel each.
+//!
+//! The input data is similar, except that is allows the inner vectors to be `AsRef<[f32]>` or `AsRef<[f64]>`.
+//! Normal vectors can be used since `Vec` implements the `AsRef` trait.
```

Comment:
I suggest to make the README.md very short and link to https://docs.rs/rubato so you don't need to maintain the text in two places.

Reply:
I don't modify the README.md at all, instead I use [cargo-readme](https://github.com/livioribeiro/cargo-readme)  to generate it from lib.rs. That works great as long as I remember to make the changes in lib.rs and not README.md..

Reply:
Neat, I wasn't aware of that. You could setup a commit hook to do that automatically.